rocketpy/motors/tank.py

```diff
@@ -410,6 +409,73 @@ def inertia(self):
         """
         return self.liquid_inertia + self.gas_inertia
 
+    def _check_volume_bounds(self):
+        """Checks if the tank is overfilled or underfilled. Raises a ValueError
+        if the tank is overfilled or underfilled.
+        """
+        if (self.fluid_volume > self.geometry.total_volume + 1e-6).any():
+            raise ValueError(
+                f"The tank '{self.name}' is overfilled. The fluid volume is "
```

Comment:
maybe you could create a single string to be used in both error messages, thus reducing the number of lines here.

Something like this.

error_msg = "The tank '{self.name}' is {fill_condition}'. " + "Consider increasing input fluid quantities or checking the fluid density values.\n\t\t" + "The fluid volume is {volume:.3f} m³ at {time:.3f} s."

raise ValueError(error_msg.format(volume=self.fluid_volume.y_array[max_volume_idx],
            time=self.fluid_volume.x_array[max_volume_idx]))

Sorry for keeping this short.